src/many-kvstore/src/module/event.rs

```diff
@@ -93,3 +99,106 @@ fn filter_date<'a>(
         Ok(events::EventLog { time, .. }) => range.contains(time),
     }))
 }
+
+fn filter_attribute_specific<'a>(
+    mut it: Box<dyn Iterator<Item = EventLogResult> + 'a>,
+    attribute_specific: &'a BTreeMap<
+        EventFilterAttributeSpecificIndex,
+        EventFilterAttributeSpecific,
+    >,
+) -> Box<dyn Iterator<Item = EventLogResult> + 'a> {
+    for x in attribute_specific.values() {
+        match x {
+            EventFilterAttributeSpecific::MultisigTransactionState(VecOrSingle(state)) => {
+                it = Box::new(it.filter(|t| match t {
+                    Err(_) => true,
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigSubmit { .. },
+                        ..
+                    })
+                    | Ok(EventLog {
+                        content: EventInfo::AccountMultisigApprove { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Pending),
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigExecute { .. },
+                        ..
+                    }) => {
+                        state.contains(&MultisigTransactionState::ExecutedAutomatically)
+                            || state.contains(&MultisigTransactionState::ExecutedManually)
+                    }
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigWithdraw { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Withdrawn),
+                    Ok(EventLog {
+                        content: EventInfo::AccountMultisigExpired { .. },
+                        ..
+                    }) => state.contains(&MultisigTransactionState::Expired),
+                    _ => false,
+                }))
```

Comment:
Is there a way to check that the internal state is one of the `MultisigTransactionState::*` without having to list them explicitly?

Reply:
I implemented this with @hansl himself, I don't think there's a better way of doing so.